ignite/metrics/clustering/calinski_harabasz_score.py

```diff
@@ -0,0 +1,107 @@
+from typing import Any, Callable, Union
+
+import torch
+from torch import Tensor
+
+from ignite.metrics.clustering._base import _ClusteringMetricBase
+
+
+def _get_calinski_harabasz_score(**kwargs: Any) -> Callable[[Tensor, Tensor], float]:
+    from sklearn.metrics import calinski_harabasz_score
+
+    def _calinski_harabasz_score(features: Tensor, labels: Tensor) -> float:
+        np_features = features.numpy()
+        np_labels = labels.numpy()
+        score = calinski_harabasz_score(np_features, np_labels, **kwargs)
+        return score
+
+    return _calinski_harabasz_score
```

Comment:
Concerning this way of importing once and returning the computation inner method, I understand the purpose but we can follow : https://github.com/pytorch/ignite/blob/302c707f79dbef50a3920baf11b76eba5ee4200e/ignite/metrics/average_precision.py#L8-L13

I think it is not a big deal with the import called multiple times. What do you think?

Reply:
I think so too.

I compared the two ways of import and found that the difference is negligible:

```python
import time

import numpy as np


def get_metric():
    from sklearn.metrics import accuracy_score

    def _accuracy_score(y_pred, y):
        score = accuracy_score(y, y_pred)
        return score

    return _accuracy_score


start = time.perf_counter()

acc = get_metric()

for _ in range(100000):
    y_pred = np.random.randint(10, size=100)
    y = np.random.randint(10, size=100)
    acc(y_pred, y)

end = time.perf_counter()
print(end - start)
```

```python
import time

import numpy as np


def acc_score(y_pred, y):
    from sklearn.metrics import accuracy_score

    score = accuracy_score(y, y_pred)
    return score


start = time.perf_counter()

for _ in range(100000):
    y_pred = np.random.randint(10, size=100)
    y = np.random.randint(10, size=100)
    acc_score(y_pred, y)

end = time.perf_counter()
print(end - start)
```